addons/im_livechat/models/discuss_channel.py

```diff
@@ -178,7 +178,8 @@ def _get_channel_history(self):
             :1
         ].script_step_id.chatbot_script_id.operator_partner_id
         last_msg_from_chatbot = False
-        for message in (self.message_ids - self.message_ids._filter_empty()).sorted("id"):
+        # sudo - mail.message: getting empty messages to exclude them is allowed.
+        for message in (self.message_ids - self.message_ids.sudo()._filter_empty()).sorted("id"):
```

Comment:
I wonder if the sudo shouldn't be in `_filter_empty` directly, as its a low level method that should work regardless of ACL.